### PR TITLE
Add curriculum filter to catalog

### PR DESCRIPTION
**Pull request by rshipp**
Add another filter, "Curriculum", that allows filtering on "marketing initiative". Because this led to the text in the "Clear filters" button being cut off, I also restructured these rows a little to break it up onto multiple lines.

image

At 2x zoom:

image

In RTL mode:

image

## Links

- jira ticket: [ACQ-820](https://codedotorg.atlassian.net/browse/ACQ-820)

### CODE REVIEW

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -1487,6 +1487,12 @@
   "manageYourLibraries": "Manage Your Libraries",
   "map": "Map",
   "mapsToCSTAStandards": "Maps to [CSTA Standards]({cstaLink})",
+  "marketingInitiativeCSA": "AP CS A",
```
Comment (TurnerRiley) [inline apps/i18n/common/en_us.json]: Should we do "AP CSA" to be consistent with how it's listed elsewhere on this page as "CSA"?

File: apps/style/code-studio/curriculum_catalog_filters.module.scss [hunk 2]
```diff
@@ -47,3 +60,9 @@ button.catalogClearFiltersButton {
   font-size: 16px;
   margin-left: auto;
 }
+
+[dir="rtl"] button.catalogClearFiltersButton {
+  font-size: 16px;
+  margin-right: auto;
+  margin-left: 0;
+}
```
Comment (TurnerRiley) [inline apps/style/code-studio/curriculum_catalog_filters.module.scss:68]: Thank you for adding this styling!!

**Review by TurnerRiley — APPROVED** [review]
This looks great! One other thing I'm noticing is that filtering for CSA, CSP, and CSD is yielding no results (I'm assuming @dmcavoy that would just require updating the `marketing_initiative` field in the course offering editor?)

**Comment by dmcavoy** [comment]
@TurnerRiley have you seeded your local environment recently? I did go through and set the marketing initiative for all CSP, CSD, and CSA courses in the last couple weeks.

**Comment by dmcavoy** [comment]
I see the things I expect when I pull this locally. I have some asks out to the Curriculum team to check the tagging but that should not block this PR going out. They can always update after launch.

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (rshipp) [reply apps/i18n/common/en_us.json]: good eye, I pulled this from the figma and missed this inconsistency.

**Comment by TurnerRiley** [comment]
> @TurnerRiley have you seeded your local environment recently? I did go through and set the marketing initiative for all CSP, CSD, and CSA courses in the last couple weeks.

Oh yep that's probably the issue my bad!

**Comment by TurnerRiley** [comment]
> @TurnerRiley have you seeded your local environment recently? I did go through and set the marketing initiative for all CSP, CSD, and CSA courses in the last couple weeks.

Oh yep that's probably the issue my bad!

**Pull request merged**